### PR TITLE
Add Talromur2 recipe

### PR DESCRIPTION
**Pull request by G-Thor**
This adds a recipe for the [Talrómur 2 multi-speaker corpus](https://repository.clarin.is/repository/xmlui/handle/20.500.12537/167). I've trained an [x-vector conditioned Tacotron 2](https://huggingface.co/espnet/talromur2_xvector_tacotron2) using this recipe with decent results.

The commit history is a bit messy so feel free to squash if you decide to merge this PR.

I'm open to any and all comments on how to improve this recipe.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -83,3 +83,7 @@ tools/anaconda
 tools/ice-g2p
 tools/fairseq
 tools/._*
+tools/anaconda
+tools/ice-g2p*
+tools/fairseq*
+tools/featbin*
```
Comment (kan-bayashi) [inline .gitignore:89]: Please add line break

File: egs2/talromur2/tts1/local/data_download.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    unzip *.zip
+    rm ./*.zip
+    touch talromur2.done
+else
+    echo "Data already present. Skipping..."
+fi
+
+# TODO(G-Thor) add alignment download option
+
+cd "${cwd}"
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/local/data_download.sh:33]: please add line break.

File: egs2/talromur2/tts1/local/data_multi_speaker.sh [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    mv ${text}_sorted ${text}
+    echo "Validating data directory"
+    utils/fix_data_dir.sh data/${full_set}
+    utils/validate_data_dir.sh --no-feats data/${full_set}
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "stage 2: utils/subset_data_dir.sh"
+
+    ./local/split_train_dev_test.py --data_dir "data/${full_set}" --train_dir "data/${train_set}" --dev_dir "data/${dev_set}" --test_dir "data/${eval_set}"
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/local/data_multi_speaker.sh:112]:
```suggestion
    ./local/split_train_dev_test.py \
        --data_dir "data/${full_set}" \
        --train_dir "data/${train_set}" \
        --dev_dir "data/${dev_set}" \
        --test_dir "data/${eval_set}"
```

File: egs2/talromur2/tts1/local/phonetize.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+. ./cmd.sh
+. ./path.sh
+
+for dset in train dev eval1; do
+    utils/copy_data_dir.sh data/"${dset}"{,_phn};
+    ${train_cmd} --gpu 1 --num-threads 1 data/"${dset}_phn/log/conversion.log" ./pyscripts/utils/convert_text_to_phn.py --nj 1 --g2p g2p_is data/"${dset}"{,_phn}/text;
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/local/phonetize.sh:8]: I think `--num-threads 1` is default.
```suggestion
    ${train_cmd} --gpu 1 data/"${dset}_phn/log/conversion.log" \
        ./pyscripts/utils/convert_text_to_phn.py \
            --nj 1 \
            --g2p g2p_is \
            data/"${dset}"{,_phn}/text
```

File: egs2/talromur2/tts1/local/phonetize.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    # srun --gres=gpu:1 ./pyscripts/utils/convert_text_to_phn.py --nj 1 --g2p g2p_is --cleaner tacotron data/"${dset}"{,_phn}/text;
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/local/phonetize.sh:9]:
```suggestion
```

File: egs2/talromur2/tts1/local/phonetize.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+done
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/local/phonetize.sh:10]: Please add line break.

File: egs2/talromur2/tts1/run.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    --train_config "${train_config}" \
+    --inference_config "${inference_config}" \
+    ${opts} "$@"
+
+## Need to supply the following values in call to run.sh
+    # --train_set "${train_set}" \
+    # --valid_set "${valid_set}" \
+    # --test_sets "${test_sets}" \
+    # --expdir "${expdir}" \
+    # --srctexts "data/${train_set}/text" \
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/run.sh:43]: Please add line break.

File: egs2/talromur2/tts1/train_multi_speaker_tacotron2.sh [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    --srctexts "data/${train_set}_phn/text" \
+    --expdir "$expdir" \
+    --stage 2 \
+    --g2p none \
+    --cleaner none \
+    --use_xvector true \
+    --ngpu 1 \
+    --expdir "$expdir" \
+    --train_config ./conf/tuning/train_xvector_tacotron2.yaml \
+    --inference_model valid.loss.ave_5best.pth
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/train_multi_speaker_tacotron2.sh:36]: Please add line break.

File: egs2/talromur2/tts1/train_multi_speaker_VITS.sh [hunk 9]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
```
Comment (kan-bayashi) [inline egs2/talromur2/tts1/train_multi_speaker_VITS.sh:1]: This is just my preference, but could use lower case? VITS -> vits

**Review by kan-bayashi — CHANGES_REQUESTED** [review]
Sorry for the late review and thank you for adding a great recipe!
It looks almost perfect but I just left minor comments. Could you reflect them?

**Comment by kan-bayashi** [comment]
Hi @G-Thor, we want to merge your great recipe.
You may struggle with ICASSP but could you reflect my review?

**Comment by G-Thor** [comment]
Hi @kan-bayashi, thanks for your review. I've been on leave and waiting for updates to the official corpus repo.
I've now applied your suggested changes as well as removing some unused code.

I also went ahead and changed the installation of ice-g2p to use the official PyPI version of that package rather than my personal fork since my suggested changes to that project have been merged. I hope it is okay to apply this change here, but if it isn't, just lmk and I'll open a separate PR for that.

**Review by kan-bayashi — APPROVED** [review]
LGTM
Thank you for your great recipe :)

**Pull request merged**